server/src/main/java/org/elasticsearch/node/NodeConstruction.java

```diff
@@ -497,6 +497,13 @@ private SettingsModule validateSettings(Settings envSettings, Settings settings,
         for (final ExecutorBuilder<?> builder : threadPool.builders()) {
             additionalSettings.addAll(builder.getRegisteredSettings());
         }
+        // TODO remove the below settings, they are unused and only here to enable BwC for deployments that still use them
```

Comment:
Nit: what about wrapping these in a function and annotate it with `@UpdateForV9` so we remember to remove them?

Reply:
done :)